contracts/adapters/TributeNFT.sol

```diff
@@ -180,6 +189,30 @@ contract TributeNFTContract is
                 UNITS,
                 proposal.requestAmount
             );
+        }
+        //if proposal passes and its an erc1155 token - use ERC1155TokenExtension
+        else if (
+            voteResult == IVoting.VotingState.PASS && proposal.tributeAmount > 0
+        ) {
+            ERC1155TokenExtension erc1155Ext =
+                ERC1155TokenExtension(dao.getExtensionAddress(ERC1155_EXT));
+            BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+            require(
+                bank.isInternalToken(UNITS),
+                "UNITS token is not an internal token"
+            );
+
+            erc1155Ext.collect(
+                proposal.applicant,
+                proposal.nftAddr,
+                proposal.nftTokenId,
+                proposal.tributeAmount
+            );
+            bank.addToBalance(
+                proposal.applicant,
+                UNITS,
+                proposal.requestAmount
+            );
```

Comment:
It looks good. I was wondering if we could do the else/if only for the `collect` call.
Something like:
```
require(bank.isInternalToken(UNITS), "UNITS token is not an internal token");

if (proposal.tributeAmount > 0) {
   ERC1155TokenExtension erc1155Ext =
                ERC1155TokenExtension(dao.getExtensionAddress(ERC1155_EXT));
   erc1155Ext.collect(
                proposal.applicant,
                proposal.nftAddr,
                proposal.nftTokenId,
                proposal.tributeAmount
            );
} else {
   NFTExtension nftExt = NFTExtension(dao.getExtensionAddress(NFT));
   nftExt.collect(proposal.nftAddr, proposal.nftTokenId);
}

bank.addToBalance...

```

wdyt?

Reply:
I agree that looks more efficient.